ui-speedspacechart/src/components/utils.ts

```diff
@@ -63,26 +63,34 @@ export const getAdaptiveHeight = (
   layersDisplay: Store['layersDisplay'],
   isIncludingLinearLayers: boolean = true
 ): number => {
-  const { ELECTRICAL_PROFILES_HEIGHT, POWER_RESTRICTIONS_HEIGHT, SPEED_LIMIT_TAGS_HEIGHT } =
-    LAYERS_HEIGHTS;
-
-  const layerHeights = {
-    electricalProfiles: ELECTRICAL_PROFILES_HEIGHT,
-    powerRestrictions: POWER_RESTRICTIONS_HEIGHT,
-    speedLimitTags: SPEED_LIMIT_TAGS_HEIGHT,
-  };
-
   let adjustment = 0;
 
-  Object.keys(layerHeights).forEach((key) => {
-    const layer = key as keyof typeof layerHeights;
+  Object.keys(LinearLayerHeights).forEach((key) => {
+    const layer = key as keyof typeof LinearLayerHeights;
     if (layersDisplay[layer]) {
-      adjustment += isIncludingLinearLayers ? layerHeights[layer] : -layerHeights[layer];
+      adjustment += isIncludingLinearLayers
+        ? LinearLayerHeights[layer]
+        : -LinearLayerHeights[layer];
     }
   });
   return height + adjustment;
 };
 
+/**
+ * Calculates the top position for the linear layers based on the height and the layers displayed.
+ */
+export const getLinearLayerMarginTop = (height: number, layersDisplay: Store['layersDisplay']) => {
+  let adjustment = 0;
+  const { electricalProfiles, powerRestrictions, speedLimitTags } = layersDisplay;
+
+  if (electricalProfiles && (powerRestrictions || speedLimitTags))
+    adjustment = LinearLayerHeights.electricalProfiles;
+
+  if (powerRestrictions && speedLimitTags) adjustment += LinearLayerHeights.powerRestrictions;
+
+  return height + adjustment - MARGINS.MARGIN_BOTTOM;
+};
```

Comment:
Hmm, I'm not sure I'm following what this function is doing…

It seems like this function is computing the top margin specifically for the power restrictions layer? I don't think we'd be able to use it for other layers, since the top margin depends on the layer stacking order. If so, maybe it would be worth indicating in the function name that it's specific to power restrictions?

Second, I'm not sure I understand the conditions. `powerRestrictions` will always be true for the result of the function to be actually used by the caller. And ultimately I think we only care about which layer stacked on top of the power restrictions row is enabled?

It seems like the following code works fine in my (limited) testing:

```js
export const getLinearLayerMarginTop = (height: number, layersDisplay: Store['layersDisplay']) => {
  let adjustment = 0;
  const { electricalProfiles, powerRestrictions, speedLimitTags } = layersDisplay;

  if (electricalProfiles)
    adjustment += LinearLayerHeights.electricalProfiles;

  if (speedLimitTags) adjustment += LinearLayerHeights.speedLimitTags;

  return height + adjustment - MARGINS.MARGIN_BOTTOM;
};
```

Am I missing something here?

Reply:
Good catch ! Done by adding a third boolean parameter for the speed limit tags linear layer